cnf-certification-test/lifecycle/suite.go

```diff
@@ -634,7 +634,7 @@ func testPodsRecreation(check *checksdb.Check, env *provider.TestEnvironment) {
 	}
 
 	for nodeName := range podsets.GetAllNodesForAllPodSets(env.Pods) {
-		defer podrecreation.CordonCleanup(nodeName) //nolint:gocritic // The defer in loop is intentional, calling the cleanup function once per node
+		defer podrecreation.CordonCleanup(nodeName, check) //nolint:gocritic // The defer in loop is intentional, calling the cleanup function once per node
```

Comment:
I'm not sure this is a good idea. The code in this checkFn runs in a separate goroutine. Sending the abort signal won't stop the execution of the rest of deferred funcs, which might take some (unknown time). In parallel, the reader will run the OnAbort() on the remaining test suites and will exit the program execution. If I'm not wrong, this means that the main goroutine might make the program to finish before all the deferred funcs here have finished.

Reply:
I merged #1687 before this.

I've been playing around with just adding a `panic()` in the execution and it seems to give a full blown panic when I run the `pods-recreation` test:

![image](https://github.com/test-network-function/cnf-certification-test/assets/4563082/10fec39d-3892-427c-a7bc-c4206ed82c0d)


![image](https://github.com/test-network-function/cnf-certification-test/assets/4563082/1192148d-b846-48ee-a405-9989f457d724)

But it seems to gracefully finish out the suite in the logs:

![image](https://github.com/test-network-function/cnf-certification-test/assets/4563082/09e5d400-879b-44cd-aa00-f131ad75d135)

What do you think about just adding the panic?  Could be a simple solution?

Reply:
Yeah, panic is (part of) the solution, but we need to detect it and recover from it to avoid showing the stacktrace. I found a way to implement that on top of this change, so I believe you can merge this as it is right now.